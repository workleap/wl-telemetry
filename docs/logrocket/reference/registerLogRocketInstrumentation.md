---
order: 100
label: registerLogRocketInstrumentation
meta:
    title: registerLogRocketInstrumentation - LogRocket
toc:
    depth: 2-3
---

# registerLogRocketInstrumentation

Initializes [LogRocket](https://logrocket.com/) instrumentation with Workleap's default settings.

## Reference

```ts
const client = registerLogRocketInstrumentation(appId, options?: { rootHostname, privateFieldNames, privateQueryParameterNames })
```

### Parameters

- `appId`: The LogRocket application id.
- `options`: An optional object literal of [predefined options](#predefined-options).

### Returns

A [LogRocketInstrumentationClient](./LogRocketInstrumentationClient.md) instance.

## Predefined options

The `registerLogRocketInstrumentation(appId, options?: {})` function also accepts a few predefined options ðŸ‘‡

### `rootHostname`

- **Type**: `string`
- **Default**: `workleap.com`

A [root hostname](https://docs.logrocket.com/reference/roothostname) to track sessions across subdomains.

```ts !#4
import { registerLogRocketInstrumentation } from "@workleap/logrocket/react";

const client = registerLogRocketInstrumentation("my-app-id", {
    rootHostname: "an-host.com"
});
```

### `privateFieldNames`

- **Type**: `string[]`
- **Default**:  `undefined`

Names of additional fields to exclude from session replays. These fields will be removed from network requests, responses using a fuzzy-matching algorithm.

```ts !#4
import { registerLogRocketInstrumentation } from "@workleap/logrocket/react";

const client = registerLogRocketInstrumentation("my-app-id", {
    privateFieldNames: ["a-custom-field"]
});
```

To view the default private fields, have a look at the [registerLogRocketInstrumentation.ts](https://github.com/workleap/wl-telemetry/blob/main/packages/logrocket/src/registerLogRocketInstrumentation.ts) file on GitHub.

### `privateQueryParameterNames`

- **Type**: `string[]`
- **Default**:  `undefined`

Names of additional fields to exclude from session replays. These fields will be removed from query parameters using a fuzzy-matching algorithm.

```ts !#4
import { registerLogRocketInstrumentation } from "@workleap/logrocket/react";

const client = registerLogRocketInstrumentation("my-app-id", {
    privateQueryParameterNames: ["a-custom-param"]
});
```

To view the default private query parameters, have a look at the [registerLogRocketInstrumentation.ts](https://github.com/workleap/wl-telemetry/blob/main/packages/logrocket/src/registerLogRocketInstrumentation.ts) file on GitHub.

### `telemetryContext`

- **Type**: `TelemetryContext`
- **Default**: `undefined`

A [TelemetryContext](../../introduction/reference/TelemetryContext.md) instance containing the telemetry correlation ids to attach to LogRocket session replays. Starting with version `2.0`, if no telemetry context is provided, the correlation ids will not be attached to LogRocket session replays.

```ts !#3,6
import { registerLogRocketInstrumentation, createTelemetryContext } from "@workleap/logrocket/react";

const telemetryContext = createTelemetryContext();

const client = registerLogRocketInstrumentation("my-app-id", {
    telemetryContext
});
```

### `verbose`

- **Type**: `boolean`
- **Default**: `false`

If no `loggers` are configured, verbose mode will automatically send logs to the console. In some cases, enabling verbose mode also produces additional debug information.

```ts !#4
import { registerLogRocketInstrumentation } from "@workleap/logrocket/react";

const client = registerLogRocketInstrumentation("my-app-id", {
    verbose: true
});
```

### `loggers`

- **Type**: `RootLogger[]`
- **Default**: `undefined`

The logger instances that will output messages.

```ts !#5
import { registerLogRocketInstrumentation, LogRocketLogger } from "@workleap/logrocket/react";
import { BrowserConsoleLogger, LogLevel } from "@workleap/logging";

const client = registerLogRocketInstrumentation("my-app-id", {
    loggers: [new BrowserConsoleLogger(), new LogRocketLogger({ logLevel: LogLevel.information })]
});
```

## Configuration transformers

!!!warning
We do not guarantee that your configuration transformers won't break after an update. It's your responsibility to keep them up to date with new releases.
!!!

The [predefined options](#predefined-options) are useful to quickly customize the default configuration of the [LogRocket SDK](https://docs.logrocket.com/reference/init), but only covers a subset of the options. If you need full control over the configuration, you can provide configuration transformer functions through the `transformers` option of the `registerLogRocketInstrumentation` function. Remember, **no locked in** :heart::v:.

To view the default configuration of `registerLogRocketInstrumentation`, have a look at the [registerLogRocketInstrumentation.ts](https://github.com/workleap/wl-telemetry/blob/main/packages/logrocket/src/registerLogRocketInstrumentation.ts) file on GitHub.

### `transformers`

- **Type**: `((options: LogRocketSdkOptions, context: LogRocketSdkOptionsTransformer) => LogRocketSdkOptions)[]`
- **Default**: `[]`

```ts
transformer(options: LogRocketSdkOptions, context: LogRocketSdkOptionsTransformer) => LogRocketSdkOptions;
```

```ts !#3-8,11
import { registerLogRocketInstrumentation, type LogRocketSdkOptionsTransformer } from "@workleap/logrocket/react";

const disableConsoleLogging: LogRocketSdkOptionsTransformer = config => {
    config.console = ...(config.console || {});
    config.console.isEnabled = false;

    return config;
};

const client = registerLogRocketInstrumentation("my-app-id", {
    transformers: [disableConsoleLogging]
});
```

### Execution context

Generic transformers can use the `context` argument to gather additional information about their execution context:

```ts !#4,8 transformer.js
import type { LogRocketSdkOptionsTransformer } from "@workleap/logrocket/react";

const disableConsoleLogging: LogRocketSdkOptionsTransformer = (config, context) => {
    if (!context.verbose) {
        config.console = ...(config.console || {});
        config.shouldDebugLog = false;

        context.logger.debug("Disabling LogRocket SDK debug logs.");
    }

    return config;
}
```

- `verbose`: `boolean`
- `logger`: `Logger`
