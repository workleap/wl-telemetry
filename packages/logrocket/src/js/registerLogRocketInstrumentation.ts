import { createCompositeLogger, type RootLogger } from "@workleap/logging";
import { BootstrappingStore, TelemetryContext } from "@workleap/telemetry";
import LogRocket from "logrocket";
import LogrocketFuzzySanitizer from "logrocket-fuzzy-search-sanitizer";
import { applyTransformers, type LogRocketSdkOptionsTransformer } from "./applyTransformers.ts";
import { createRequestSanitizer } from "./createRequestSanitizer.ts";
import { createResponseSanitizer } from "./createResponseSanitizer.ts";
import { createUrlSanitizer } from "./createUrlSanitizer.ts";
import { HasExecutedGuard } from "./HasExecutedGuard.ts";
import { DeviceIdTrait, LogRocketInstrumentationClient, TelemetryIdTrait } from "./LogRocketInstrumentationClient.ts";
import type { LogRocketSdkOptions } from "./logRocketTypes.ts";

const DefaultPrivateFieldNames = [
    "email",
    "personalEmail",
    "phoneNumber",
    "password",
    "firstName",
    "lastName",
    "givenName",
    "fullName",
    "gender",
    "timeZone",
    "birthday"
];

const DefaultPrivateQueryParameterNames = [
    "token",
    "email",
    "password"
];

///////////////////////////

/**
 * @see {@link https://workleap.github.io/wl-telemetry}
 */
export interface RegisterLogRocketInstrumentationOptions {
    /**
     * A root hostname to track sessions across subdomains. Set this option to capture traffic from all subdomains under one session: https://docs.logrocket.com/reference/roothostname.
     */
    rootHostname?: LogRocketSdkOptions["rootHostname"];
    /**
     * Names of additional fields to exclude from session replays. These fields will be removed from network requests, responses using a fuzzy-matching algorithm.
     */
    privateFieldNames?: string[];
    /**
     * Names of additional fields to exclude from session replays. These fields will be removed from query parameters using a fuzzy-matching algorithm.
     */
    privateQueryParameterNames?: string[];
    /**
     * Indicates whether or not debug information should be logged to the console.
     */
    verbose?: boolean;
    /**
     * The logger instances that will output messages.
     */
    loggers?: RootLogger[];
    /**
     * Hooks to transform the resulting LogRocket SDK options.
     */
    transformers?: LogRocketSdkOptionsTransformer[];
}

// The function return type is mandatory, otherwise we got an error TS4058.
export function getLogRocketSdkOptions(userOptions: RegisterLogRocketInstrumentationOptions, logger: RootLogger): LogRocketSdkOptions {
    const {
        rootHostname = "workleap.com",
        privateFieldNames = [],
        privateQueryParameterNames = [],
        verbose = false,
        transformers = []
    } = userOptions;

    const mergedPrivateFieldNames = DefaultPrivateFieldNames.concat(privateFieldNames);
    const mergedPrivateQueryParameterNames = DefaultPrivateQueryParameterNames.concat(privateQueryParameterNames);

    // The "LogrocketFuzzySearch.setup" code is awkward. For now, we prefer to play it safe and call it once here, then forward the sanitize functions
    // to the appropriate factories.
    // No clue why the "setup" method is not available. This package is shady.
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    const { requestSanitizer: fuzzyRequestSanitizer, responseSanitizer: fuzzyResponseSanitizer } = LogrocketFuzzySanitizer.setup(mergedPrivateFieldNames);

    const sdkOptions = {
        console: {
            // Prevent console logs from being visible in LogRocket session replays and leaking PII.
            // To capture logs in a session replay, use the LogRocketLogger class.
            isEnabled: false
        },
        rootHostname,
        dom: {
            textSanitizer: true,
            inputSanitizer: true
        },
        network: {
            requestSanitizer: createRequestSanitizer(fuzzyRequestSanitizer),
            responseSanitizer: createResponseSanitizer(fuzzyResponseSanitizer)
        },
        browser: {
            urlSanitizer: createUrlSanitizer(mergedPrivateQueryParameterNames)
        },
        shouldDebugLog: verbose
    } satisfies LogRocketSdkOptions;

    return applyTransformers(sdkOptions, transformers, {
        verbose,
        logger
    });
}

///////////////////////////

// DEPRECATED: Grace period ends on January 1th 2026.
export const IsRegisteredVariableName = "__WLP_LOGROCKET_INSTRUMENTATION_IS_REGISTERED__";
export const RegisterGetSessionUrlFunctionName = "__WLP_LOGROCKET_INSTRUMENTATION_REGISTER_GET_SESSION_URL_LISTENER__";

// DEPRECATED: Grace period ends on January 1th 2026.
function registerDeprecatedGlobalVariables() {
    // Indicates to the host applications that logrocket has been initialized.
    // It's useful in cases where an "add-on", like the platform widgets needs
    // to know whether or not the host application is using LogRocket.
    // While there are ways that the host application could tell to an "add-on" if
    // it's using LogRocket or not, doing it this way is transparent for the consumer,
    // which is great for DX.
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    globalThis[IsRegisteredVariableName] = true;

    // While consumers could directly call LogRocket.getSessionURL, by doing it this way,
    // it allow consumers to not take a direct dependency on the "logrocket" package.
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    globalThis[RegisterGetSessionUrlFunctionName] = listener => {
        LogRocket.getSessionURL(listener);
    };
}

///////////////////////////

let registrationGuardInstance: HasExecutedGuard | undefined;

// It's important to use a lazy singleton instead of a singleton to avoid
// any future probleme related to HMR, tree-shaking, SSR, etc..
function getRegistrationGuard() {
    if (!registrationGuardInstance) {
        registrationGuardInstance = new HasExecutedGuard();
    }

    return registrationGuardInstance;
}

// This function should only be used by tests.
export function __resetRegistrationGuard() {
    if (registrationGuardInstance) {
        registrationGuardInstance.reset();
    }
}

///////////////////////////

export class LogRocketInstrumentationRegistrator {
    register(
        appId: string,
        telemetryContext: TelemetryContext,
        // TODO: Probably replace the store with the actual client instance instead.
        // and delete permanently the store implementation.
        bootstrappingStore: BootstrappingStore,
        options: RegisterLogRocketInstrumentationOptions = {}
    ) {
        const {
            verbose = false,
            loggers = []
        } = options;

        const logger = createCompositeLogger(verbose, loggers);
        const sdkOptions = getLogRocketSdkOptions(options, logger);

        // Session starts anonymously when LogRocket.init() is called.
        LogRocket.init(appId, sdkOptions);

        // LogRocket maintains the same session even if the user starts as anonymous and later becomes identified via LogRocket.identify().
        // If LogRocket.identify is called multiple times during a recording, you can search for any of the identified users in the session.
        LogRocket.identify(telemetryContext.deviceId, {
            [DeviceIdTrait]: telemetryContext.deviceId,
            [TelemetryIdTrait]: telemetryContext.telemetryId
        });

        LogRocket.getSessionURL(url => {
            logger
                .withText("[logrocket] Session replay URL is now available:")
                .withText(url)
                .debug();
        });

        registerDeprecatedGlobalVariables();

        // Let the other telemetry libraries know that LogRocket instrumentation is ready.
        bootstrappingStore.dispatch({ type: "logrocket-ready" });

        logger.information("[logrocket] LogRocket instrumentation is registered.");

        return new LogRocketInstrumentationClient(telemetryContext);
    }
}

/**
 * @see {@link https://workleap.github.io/wl-telemetry}
 */
export function registerLogRocketInstrumentation(
    appId: string,
    telemetryContext: TelemetryContext,
    // TODO: Probably replace the store with the actual client instance instead.
    // and delete permanently the store implementation.
    bootstrappingStore: BootstrappingStore,
    options?: RegisterLogRocketInstrumentationOptions
) {
    getRegistrationGuard().throw("[logrocket] The LogRocket instrumentation has already been registered. Did you call the \"registerLogRocketInstrumentation\" function twice?");

    return new LogRocketInstrumentationRegistrator().register(appId, telemetryContext, bootstrappingStore, options);
}
