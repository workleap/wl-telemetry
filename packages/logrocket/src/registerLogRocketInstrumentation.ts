import { createCompositeLogger, type RootLogger } from "@workleap/logging";
import { createBootstrappingStore, createTelemetryContext } from "@workleap/telemetry";
import LogRocket from "logrocket";
import LogrocketFuzzySanitizer from "logrocket-fuzzy-search-sanitizer";
import { applyTransformers, type LogRocketSdkOptionsTransformer } from "./applyTransformers.ts";
import { DeviceIdTrait, TelemetryIdTrait } from "./createDefaultUserTraits.ts";
import { createRequestSanitizer } from "./createRequestSanitizer.ts";
import { createResponseSanitizer } from "./createResponseSanitizer.ts";
import { createUrlSanitizer } from "./createUrlSanitizer.ts";
import { HasExecutedGuard } from "./HasExecutedGuard.ts";
import type { LogRocketSdkOptions } from "./logRocketTypes.ts";

export const IsRegisteredVariableName = "__WLP_LOGROCKET_INSTRUMENTATION_IS_REGISTERED__";
export const RegisterGetSessionUrlFunctionName = "__WLP_LOGROCKET_INSTRUMENTATION_REGISTER_GET_SESSION_URL_LISTENER__";

const DefaultPrivateFieldNames = [
    "email",
    "personalEmail",
    "phoneNumber",
    "password",
    "firstName",
    "lastName",
    "givenName",
    "fullName",
    "gender",
    "timeZone",
    "birthday"
];

const DefaultPrivateQueryParameterNames = [
    "token",
    "email",
    "password"
];

/**
 * @see {@link https://workleap.github.io/wl-telemetry}
 */
export interface RegisterLogRocketInstrumentationOptions {
    /**
     * A root hostname to track sessions across subdomains. Set this option to capture traffic from all subdomains under one session: https://docs.logrocket.com/reference/roothostname.
     */
    rootHostname?: LogRocketSdkOptions["rootHostname"];
    /**
     * Names of additional fields to exclude from session replays. These fields will be removed from network requests, responses using a fuzzy-matching algorithm.
     */
    privateFieldNames?: string[];
    /**
     * Names of additional fields to exclude from session replays. These fields will be removed from query parameters using a fuzzy-matching algorithm.
     */
    privateQueryParameterNames?: string[];
    /**
     * Indicates whether or not debug information should be logged to the console.
     */
    verbose?: boolean;
    /**
     * The logger instances that will output messages.
     */
    loggers?: RootLogger[];
    /**
     * Hooks to transform the resulting LogRocket SDK options.
     */
    transformers?: LogRocketSdkOptionsTransformer[];
}

// The function return type is mandatory, otherwise we got an error TS4058.
export function getLogRocketSdkOptions(userOptions: RegisterLogRocketInstrumentationOptions, logger: RootLogger): LogRocketSdkOptions {
    const {
        rootHostname = "workleap.com",
        privateFieldNames = [],
        privateQueryParameterNames = [],
        verbose = false,
        transformers = []
    } = userOptions;

    const mergedPrivateFieldNames = DefaultPrivateFieldNames.concat(privateFieldNames);
    const mergedPrivateQueryParameterNames = DefaultPrivateQueryParameterNames.concat(privateQueryParameterNames);

    // The "LogrocketFuzzySearch.setup" code is awkward. For now, we prefer to play it safe and call it once here, then forward the sanitize functions
    // to the appropriate factories.
    // No clue why the "setup" method is not available. This package is shady.
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    const { requestSanitizer: fuzzyRequestSanitizer, responseSanitizer: fuzzyResponseSanitizer } = LogrocketFuzzySanitizer.setup(mergedPrivateFieldNames);

    const sdkOptions = {
        console: {
            // Prevent console logs from being visible in LogRocket session replays and leaking PII.
            // To capture logs in a session replay, use the LogRocketLogger class.
            isEnabled: false
        },
        rootHostname,
        dom: {
            textSanitizer: true,
            inputSanitizer: true
        },
        network: {
            requestSanitizer: createRequestSanitizer(fuzzyRequestSanitizer),
            responseSanitizer: createResponseSanitizer(fuzzyResponseSanitizer)
        },
        browser: {
            urlSanitizer: createUrlSanitizer(mergedPrivateQueryParameterNames)
        },
        shouldDebugLog: verbose
    } satisfies LogRocketSdkOptions;

    return applyTransformers(sdkOptions, transformers, {
        verbose,
        logger
    });
}

const registrationGuard = new HasExecutedGuard();

// This function should only be used by tests.
export function __resetRegistrationGuard() {
    registrationGuard.reset();
}

/**
 * @see {@link https://workleap.github.io/wl-telemetry}
 */
export function registerLogRocketInstrumentation(appId: string, options: RegisterLogRocketInstrumentationOptions = {}) {
    const {
        verbose = false,
        loggers = []
    } = options;

    const logger = createCompositeLogger(verbose, loggers);

    registrationGuard.throw("[logrocket] The LogRocket instrumentation has already been registered. Did you call the \"registerLogRocketInstrumentation\" function twice?");

    const sdkOptions = getLogRocketSdkOptions(options, logger);

    // Session starts anonymously when LogRocket.init() is called.
    LogRocket.init(appId, sdkOptions);

    const telemetryContext = createTelemetryContext(logger);

    // LogRocket maintains the same session even if the user starts as anonymous and later becomes identified via LogRocket.identify().
    // If LogRocket.identify is called multiple times during a recording, you can search for any of the identified users in the session.
    LogRocket.identify(telemetryContext.deviceId, {
        [DeviceIdTrait]: telemetryContext.deviceId,
        [TelemetryIdTrait]: telemetryContext.telemetryId
    });

    LogRocket.getSessionURL(url => {
        logger
            .withText("[logrocket] Session replay URL is now available:")
            .withText(url)
            .debug();
    });

    // Indicates to the host applications that logrocket has been initialized.
    // It's useful in cases where an "add-on", like the platform widgets needs
    // to know whether or not the host application is using LogRocket.
    // While there are ways that the host application could tell to an "add-on" if
    // it's using LogRocket or not, doing it this way is transparent for the consumer,
    // which is great for DX.
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    globalThis[IsRegisteredVariableName] = true;

    // While consumers could directly call LogRocket.getSessionURL, by doing it this way,
    // it allow consumers to not take a direct dependency on the "logrocket" package.
    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
    // @ts-ignore
    globalThis[RegisterGetSessionUrlFunctionName] = listener => {
        LogRocket.getSessionURL(listener);
    };

    const bootstrappingStore = createBootstrappingStore(logger);
    // Let the other telemetry libraries know that LogRocket instrumentation is ready.
    bootstrappingStore.dispatch({ type: "logrocket-ready" });

    logger.information("[logrocket] LogRocket instrumentation is registered.");
}
