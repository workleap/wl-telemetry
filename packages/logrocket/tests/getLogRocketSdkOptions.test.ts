import { NoopLogger } from "@workleap/logging";
import { test, vi } from "vitest";
import { getLogRocketSdkOptions } from "../src/registerLogRocketInstrumentation.ts";

vi.mock("logrocket", () => ({
    default: {
        init: vi.fn(),
        identify: vi.fn(),
        log: vi.fn(),
        track: vi.fn(),
        getSessionURL: vi.fn()
    }
}));

test.concurrent("when a rootHostname is provided", ({ expect }) => {
    const result = getLogRocketSdkOptions({
        rootHostname: "toto"
    }, new NoopLogger());

    expect(result).toMatchSnapshot();
});

test.concurrent("when privateFieldNames are provided", ({ expect }) => {
    const result = getLogRocketSdkOptions({
        privateFieldNames: ["foo", "bar"]
    }, new NoopLogger());

    expect(result).toMatchSnapshot();
});

test.concurrent("when privateQueryParameterNames are provided", ({ expect }) => {
    const result = getLogRocketSdkOptions({
        privateQueryParameterNames: ["foo", "bar"]
    }, new NoopLogger());

    expect(result).toMatchSnapshot();
});

test.concurrent("when verbose is true", ({ expect }) => {
    const result = getLogRocketSdkOptions({
        verbose: true
    }, new NoopLogger());

    expect(result).toMatchSnapshot();
});

test.concurrent("with a single transformer", ({ expect }) => {
    const result = getLogRocketSdkOptions({
        transformers: [
            options => {
                options.rootHostname = "toto";

                return options;
            }
        ]
    }, new NoopLogger());

    expect(result).toMatchSnapshot();
});

test.concurrent("with a multiple transformers", ({ expect }) => {
    const result = getLogRocketSdkOptions({
        transformers: [
            options => {
                options.rootHostname = "toto";

                return options;
            },
            options => {
                options.serverURL = "foo";

                return options;
            }
        ]
    }, new NoopLogger());

    expect(result).toMatchSnapshot();
});
