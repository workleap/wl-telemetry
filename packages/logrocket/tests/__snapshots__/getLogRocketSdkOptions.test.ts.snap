// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`when a rootHostname is provided 1`] = `
{
  "browser": {
    "urlSanitizer": [Function],
  },
  "console": {
    "isEnabled": false,
  },
  "dom": {
    "inputSanitizer": true,
    "textSanitizer": true,
  },
  "network": {
    "requestSanitizer": [Function],
    "responseSanitizer": [Function],
  },
  "rootHostname": "toto",
  "shouldDebugLog": false,
}
`;

exports[`when privateFieldNames are provided 1`] = `
{
  "browser": {
    "urlSanitizer": [Function],
  },
  "console": {
    "isEnabled": false,
  },
  "dom": {
    "inputSanitizer": true,
    "textSanitizer": true,
  },
  "network": {
    "requestSanitizer": [Function],
    "responseSanitizer": [Function],
  },
  "rootHostname": "workleap.com",
  "shouldDebugLog": false,
}
`;

exports[`when privateQueryParameterNames are provided 1`] = `
{
  "browser": {
    "urlSanitizer": [Function],
  },
  "console": {
    "isEnabled": false,
  },
  "dom": {
    "inputSanitizer": true,
    "textSanitizer": true,
  },
  "network": {
    "requestSanitizer": [Function],
    "responseSanitizer": [Function],
  },
  "rootHostname": "workleap.com",
  "shouldDebugLog": false,
}
`;

exports[`when verbose is true 1`] = `
{
  "browser": {
    "urlSanitizer": [Function],
  },
  "console": {
    "isEnabled": false,
  },
  "dom": {
    "inputSanitizer": true,
    "textSanitizer": true,
  },
  "network": {
    "requestSanitizer": [Function],
    "responseSanitizer": [Function],
  },
  "rootHostname": "workleap.com",
  "shouldDebugLog": true,
}
`;

exports[`with a multiple transformers 1`] = `
{
  "browser": {
    "urlSanitizer": [Function],
  },
  "console": {
    "isEnabled": false,
  },
  "dom": {
    "inputSanitizer": true,
    "textSanitizer": true,
  },
  "network": {
    "requestSanitizer": [Function],
    "responseSanitizer": [Function],
  },
  "rootHostname": "toto",
  "serverURL": "foo",
  "shouldDebugLog": false,
}
`;

exports[`with a single transformer 1`] = `
{
  "browser": {
    "urlSanitizer": [Function],
  },
  "console": {
    "isEnabled": false,
  },
  "dom": {
    "inputSanitizer": true,
    "textSanitizer": true,
  },
  "network": {
    "requestSanitizer": [Function],
    "responseSanitizer": [Function],
  },
  "rootHostname": "toto",
  "shouldDebugLog": false,
}
`;
